src/BasicStepperDriver.h

```diff
@@ -218,5 +218,17 @@ class BasicStepperDriver {
     long calcStepsForRotation(double deg){
         return deg * motor_steps * microsteps / 360;
     }
+    /*
+     * Return steps remaining for a specific movement
+     */
+    long getStepsRemaining(){
+        return steps_remaining;
+    }
+    /*
+     * Return the movement direction
+     */
+    bool getDir(){
+        return dir_state;
```

Comment:
The first method is pretty clear, but `getDir()` returns some value (0 or 1) that represents internal state, so the caller does not know how to interpret without reading the code. For example, consider a driver implementation that reverses the logic on this pin.

You mentioned that you found these methods useful, how do you use it ? Would something like this work ?
```suggestion
        return (dir_state == HIGH) ? 1 : -1;
```
it returns 1 for forward and -1 for reverse, which is consistent with move() parameters.